testsuite/features/step_definitions/command_steps.rb

```diff
@@ -796,9 +796,17 @@
   node.run(cmd, true, DEFAULT_TIMEOUT, 'root', [0, 100, 101, 102, 103, 104, 106])
 end
 
-When(/^I (install|remove) the traditional stack utils (on|from) "([^"]*)"$/) do |action, where, host|
```

Comment:
```
When(/^I (install|remove) the traditional stack utils (on|from) "([^"]*)"$/) do |action, where, host|
  pkgs = 'spacewalk-client-tools spacewalk-check spacewalk-client-setup mgr-daemon mgr-osad mgr-cfg-actions'
  step %(I refresh the metadata for "#{host}") if action == 'install'
  step %(I #{action} packages "#{pkgs}" #{where} this "#{host}")
end
```

Reply:
We do not refresh metadata before removing packages.

Reply:
Not elegant.

Reply:
I don't find elegant code duplication. And I consider it a bad practice.
If you don't probe a good reason to make this refactor mandatory, I would like that we only add the oneliner I suggested.

Reply:
I gave a whole series of good reasons in previous comment.